File: dsa-examples/go/selection-sort.go
package main

import "fmt"

func main() {
	fmt.Println(selection_sort([]int{5, 2, 9, 1, 4, 3, 8, 6, 7}))
}

func selection_sort(nums_arr []int) []int {
	nums := nums_arr
    for i := 0; i < len(nums); i++ {
        min_index := i
        for j := i + 1; j < len(nums); j++ {
            if nums[min_index] > nums[j] {
                min_index = j
			}
		}
        nums[i], nums[min_index] = nums[min_index], nums[i]
	}
	return nums
}

File: dsa-examples/python/selection-sort.py
# Selection Sort
# Time Complexity: Best: O(n^2), Average: O(n^2), Worst: O(n^2)
# Finds the lowest number in each iteration and swaps it with the current iteration number
def selection_sort(nums):

    nums = list(nums)

    for i in range(len(nums)):

        min_index = i

        for j in range(i + 1, len(nums)):

            if nums[min_index] > nums[j]:
                min_index = j

        nums[i], nums[min_index] = nums[min_index], nums[i]

    return nums
